TYPO3.Neos/Classes/TYPO3/Neos/Service/View/NodeView.php

```diff
@@ -237,7 +237,9 @@ public function collectParentNodeData(NodeInterface $rootNode, array $nodes)
                     $collectTreeNodeData($children, $childNode);
                 }
             }
-            $treeNodes[] = $self->collectTreeNodeData($node['node'], true, $children, $children !== array(), isset($node['matched']));
+            if (isset($node['node']) && $node['node'] instanceof NodeInterface) {
```

Comment:
well it must happen one way or the other, did you try doing some debugging and see if that variable is set correctly in the array?